R/migrations.R

```diff
@@ -0,0 +1,93 @@
+run_migration <- function(queue, output_dir, from_version, to_version,
+                          dry_run = TRUE) {
+  output_dir <- normalizePath(output_dir, mustWork = TRUE)
+  tasks <- queue$queue$task_list()
+  status <- queue$queue$task_status(tasks)
+  completed_tasks <- tasks[status == "COMPLETE"]
+  migrations <- lapply(completed_tasks, migrate_task, queue,
+                       from_version, to_version, dry_run)
+  summary <- lapply(migrations, function(migration) {
+    list(
+      id = migration$id,
+      action = migration$action
+    )
+  })
+  summary <- do.call(rbind, summary)
+  summary_path <- file.path(output_dir, "summary.csv")
```

Comment:
I take it it's ok to write the summary directly into the output dir? It's not going to get confused with anything else?
Is it worth timestamping the filename in case you do successive runs and want to keep previous summaries? Same for the qs file (which is the chattier log?)

Reply:
Yeah, so the calling script creates a new directory for writing out this log into. I'll tidy up this name as it is a bit confusing, will call it `log_dir` as that really is what it is. I think definitely a good idea to timestamp those files though - I'll do that too. The qs file is the chattier one yeah.